ui/build.gradle.kts

```diff
@@ -109,35 +120,39 @@ val npmInstall =
     }
   }
 
+val openapiGenerator by configurations.creating
+
+dependencies { openapiGenerator(libs.openapi.generator.cli) }
+
 val npmGenerateAPI =
-  tasks.register<NpmTask>("npmGenerateApi") {
+  tasks.register<JavaExec>("npmGenerateApi") {
     description = "Generate from OpenAPI spec"
-    dependsOn(npmInstall)
-    inputs.dir(shadowPackageJson).withPathSensitivity(PathSensitivity.RELATIVE)
+    outputs.cacheIf { true }
     inputs.dir(openApiSpecDir).withPathSensitivity(PathSensitivity.RELATIVE)
-    outputs.dir(generatedOpenApiCodeUnfixed)
+    outputs.dir(generatedOpenApiCode)
+    classpath(openapiGenerator)
+    mainClass.set("org.openapitools.codegen.OpenAPIGenerator")
+    logging.captureStandardOutput(LogLevel.INFO)
+    logging.captureStandardError(LogLevel.LIFECYCLE)
+    args(
+      "generate",
+      "-g",
+      "typescript-fetch",
+      "-i",
+      "${openApiSpecDir.relativeTo(projectDir)}/nessie-openapi-0.45.0.yaml",
+      "-o",
+      "src/generated/utils/api",
+      "--additional-properties=supportsES6=true"
+    )
     doFirst {
       // Remove previously generated code to have generated files consistent with the OpenAPI spec
-      delete(generatedOpenApiCodeUnfixed)
+      delete(generatedOpenApiCode)
     }
-    args.set(listOf("run", "generate-api"))
-  }
-
-val npmFixGeneratedClient =
-  tasks.register<NpmTask>("npmFixGeneratedClient") {
-    description = "Fix generated OpenAPI code"
-    dependsOn(npmGenerateAPI)
-    inputs.dir(generatedOpenApiCodeUnfixed).withPathSensitivity(PathSensitivity.RELATIVE)
-    outputs.dir(generatedOpenApiCode)
-    doFirst {
-      sync {
-        // "Sync" ensures that the contents in src/generated match the contents of the
-        // code generated via npmGenerateAPI
-        into(generatedOpenApiCode)
-        from(generatedOpenApiCodeUnfixed.resolve("src/generated"))
-      }
+    doLast {
+      val f = generatedOpenApiCode.resolve("utils/api/runtime.ts")
+      val src = f.readText()
+      f.writeText(src.replace("GlobalFetch", "WindowOrWorkerGlobalScope"))
```

Comment:
Could you also add the comment from the deleted `fix-generated-client.js` file here?

Reply:
done